src/plugins/azgenerator/azgenerator.ts

```diff
@@ -12,7 +12,12 @@ export async function processRequest(host: Host) {
         const session = await startSession<CodeModel>(host, {}, codeModelSchema); 
         let model = new CodeModelCliImpl(session);
         let files: any = await GenerateAll(model, true);
-
+        if (model.SelectFirstExtension()) {
+            do {
+                let path = "azext_" + model.Extension_Name.replace("-", "_") + "/";
+                session.protectFiles(path + "manual");
```

Comment:
if we have generated record for test, that should also be protected, righ?

Reply:
I think the record would be generate when we run the test not when we generate the test file. right?

Reply:
yes, but if we regenerate the code, the record shouldn't be deleted, right?

Reply:
I think if we regenerate the code. the test should also be rerun. otherwise how do we know the regenerated code is working?

Reply:
we can rerun with the recording if it's still valid, right? we only need to re-record when the recording file no longer valid, otherwise this will be painful in develop phase

Reply:
Then I can add it. but where's the recording folder @changlong-liu